libraries/standard/http/cbmc/include/http_cbmc_state.h

```diff
@@ -0,0 +1,31 @@
+#ifndef HTTP_CBMC_STATE_H_
+#define HTTP_CBMC_STATE_H_
+
+#include "http_client.h"
+
+/**
+ * @brief Calls malloc based on given size or returns NULL for coverage.
+ *
+ * Implementation of safe malloc which returns NULL if the requested size is 0.
+ * The behavior of malloc(0) is platform dependent.
+ * It is possible for malloc(0) to return an address without allocating memory.
+ *
+ * @brief param[in] wantedSize Requested size to malloc.
+ */
+void * safeMalloc( size_t wantedSize );
```

Comment:
Nicely done.  But let us checkout the cbmc flag for malloc to see if we can use that, too.

Reply:
Per offline research, we must keep using a redefined malloc.